src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SimulatorEnclaveProvider.cs

```diff
@@ -84,8 +84,11 @@ internal override void CreateEnclaveSession(byte[] attestationInfo, ECDiffieHell
                         Buffer.BlockCopy(attestationInfo, attestationInfoOffset, trustedModuleDHPublicKeySignature, 0,
                             checked((int)sizeOfTrustedModuleDHPublicKeySignatureBuffer));
 
-                        CngKey k = CngKey.Import(trustedModuleDHPublicKey, CngKeyBlobFormat.EccPublicBlob);
-                        byte[] sharedSecret = clientDHKey.DeriveKeyMaterial(k);
+                        byte[] sharedSecret;
+                        using (ECDiffieHellman ecdh = KeyConverter.CreateECDiffieHellmanFromPublicKeyBlob(trustedModuleDHPublicKey))
```

Comment:
NIT: I think using statement could be simplified after [C#8](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations).